File: server/src/main/java/com/moviekim/ansimtalk_server/medication/dto/LogResponseDto.java
package com.moviekim.ansimtalk_server.medication.dto;

import com.moviekim.ansimtalk_server.medication.MedicationLog;
import lombok.Getter;
import java.time.format.DateTimeFormatter;

@Getter
public class LogResponseDto {
    private Long logId;
    private String date;
    private boolean isTaken;
    private String takenAt; // "HH:mm" 또는 null
    private String medicationName;
    private String scheduleTime;

    public LogResponseDto(MedicationLog log) {
        this.logId = log.getId();
        this.date = log.getDate().format(DateTimeFormatter.ISO_LOCAL_DATE);
        this.isTaken = log.isTaken();
        this.takenAt = (log.getTakenAt() != null) ? log.getTakenAt().format(DateTimeFormatter.ofPattern("HH:mm")) : null;
        this.medicationName = log.getSchedule().getMedicationName();
        this.scheduleTime = log.getSchedule().getTime().format(DateTimeFormatter.ofPattern("HH:mm"));
    }
}

File: guardian-app/app/src/main/java/com/moviekim/ansimtalk/guardian/ui/dto/LogResponseDto.kt
package com.moviekim.ansimtalk.guardian.ui.dto

import com.google.gson.annotations.SerializedName

data class LogResponseDto(
    @SerializedName("logId")
    val logId: Long,

    @SerializedName("date")
    val date: String,

    @SerializedName("isTaken")
    val isTaken: Boolean,

    @SerializedName("takenAt")
    val takenAt: String?,

    @SerializedName("medicationName")
    val medicationName: String,

    @SerializedName("scheduleTime")
    val scheduleTime: String
)
